src/main/scala/org/camunda/feel/impl/builtin/ListBuiltinFunctions.scala

```diff
@@ -381,4 +385,62 @@ object ListBuiltinFunctions {
           s"expect boolean function with 2 arguments, but found '${params.size}'")
     }
   )
+
+  private def joiningFunction = builtinFunction(
+    params = List("list"),
+    invoke = {
+      case List(ValList(list)) =>
+        fromListOf[ValString, String](list)(
+          valStr => valStr.value,
+          strings => ValString(strings.mkString))
+    }
+  )
+
+  private def joiningWithDelimiterFunction = builtinFunction(
+    params = List("list", "delimiter"),
+    invoke = {
+      case List(ValList(list), ValString(delimiter)) =>
+        fromListOf[ValString, String](list)(
+          valStr => valStr.value,
+          strings => ValString(strings.mkString(delimiter)))
+    }
+  )
+
+  private def joiningWithDelimiterAndPrefixFunction = builtinFunction(
+    params = List("list", "delimiter", "prefix"),
+    invoke = {
+      case List(ValList(list), ValString(delimiter), ValString(prefix)) =>
+        fromListOf[ValString, String](list)(
+          valStr => valStr.value,
+          strings => ValString(s"$prefix${strings.mkString(delimiter)}"))
+    }
+  )
+
+  private def joiningWithDelimiterAndPrefixAndSuffixFunction = builtinFunction(
+    params = List("list", "delimiter", "prefix", "suffix"),
+    invoke = {
+      case List(ValList(list),
+                ValString(delimiter),
+                ValString(prefix),
+                ValString(suffix)) =>
+        fromListOf[ValString, String](list)(
+          valStr => valStr.value,
+          strings => ValString(s"$prefix${strings.mkString(delimiter)}$suffix"))
```

Comment:
:wrench: We could use Scala's built-in [mkString()](https://www.scala-lang.org/api/current/scala/collection/StringOps.html#mkString(start:String,sep:String,end:String):String) function with three arguments.

```suggestion
          strings => ValString(strings.mkString(start = prefix, sep = delimiter, end = suffix))
```